projects/aca-content/viewer/src/lib/services/viewer.service.ts

```diff
@@ -0,0 +1,229 @@
+/*!
+ * Copyright © 2005-2023 Hyland Software, Inc. and its affiliates. All rights reserved.
+ *
+ * Alfresco Example Content Application
+ *
+ * This file is part of the Alfresco Example Content Application.
+ * If the software was purchased under a paid Alfresco license, the terms of
+ * the paid license agreement will prevail. Otherwise, the software is
+ * provided under the following open source license terms:
+ *
+ * The Alfresco Example Content Application is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The Alfresco Example Content Application is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * from Hyland Software. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { ObjectUtils, UserPreferencesService } from '@alfresco/adf-core';
+import { FavoritePaging, Node, NodePaging, SearchRequest, ResultSetPaging, SharedLink, SharedLinkPaging } from '@alfresco/js-api';
+import { Injectable } from '@angular/core';
+import { ContentApiService } from '@alfresco/aca-shared';
+
+interface adjacentFiles {
+  left: string;
+  right: string;
+}
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ViewerService {
+  constructor(private preferences: UserPreferencesService, private contentApi: ContentApiService) {}
+
+  recentFileFilters = [
+    'TYPE:"content"',
+    '-PATH:"//cm:wiki/*"',
+    '-TYPE:"app:filelink"',
+    '-TYPE:"fm:post"',
+    '-TYPE:"cm:thumbnail"',
+    '-TYPE:"cm:failedThumbnail"',
+    '-TYPE:"cm:rating"',
+    '-TYPE:"dl:dataList"',
+    '-TYPE:"dl:todoList"',
+    '-TYPE:"dl:issue"',
+    '-TYPE:"dl:contact"',
+    '-TYPE:"dl:eventAgenda"',
+    '-TYPE:"dl:event"',
+    '-TYPE:"dl:task"',
+    '-TYPE:"dl:simpletask"',
+    '-TYPE:"dl:meetingAgenda"',
+    '-TYPE:"dl:location"',
+    '-TYPE:"fm:topic"',
+    '-TYPE:"fm:post"',
+    '-TYPE:"ia:calendarEvent"',
+    '-TYPE:"lnk:link"'
+  ];
+
+  /**
+   * Retrieves nearest node information for the given node and folder.
+   *
+   * @param nodeId Unique identifier of the document node
+   * @param folderId Unique identifier of the containing folder node.
+   * @param source Data source name. Allowed values are: personal-files, libraries, favorites, shared, recent-files.
```

Comment:
if allowed values for source are 

personal-files, libraries, favorites, shared, recent-files

then maybe we can limit type of source parameter to only these values instead of general string which allows every string?

Reply:
@AleksanderSklorz I indiscriminately reverted to old param description. Now the description is corrected, so is the getFileIds function with 'wrong' source. Your solution would be cleaner, but the source is tied to pages, and due to libraries pages' logic I believe this solution to be sufficient.